testing/unittests/jlibtests.cpp

```diff
@@ -85,6 +88,63 @@ class JlibSemTest : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( JlibSemTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( JlibSemTest, "JlibSemTest" );
 
+
+class JlibSemTestStress : public CppUnit::TestFixture
+{
+public:
+    CPPUNIT_TEST_SUITE(JlibSemTestStress);
+        CPPUNIT_TEST(testSetup);
+        CPPUNIT_TEST(testSimple);
+        CPPUNIT_TEST(testCleanup);
+    CPPUNIT_TEST_SUITE_END();
+
+protected:
+
+    void testSetup()
+    {
+    }
+
+    void testCleanup()
+    {
+    }
+
+    void testTimedElapsed(Semaphore & sem, unsigned time, unsigned loopCount)
+    {
+        unsigned long sumTaken = 0;
+        unsigned maxTaken = 0;
+        unsigned timeLimit = 2 * time;
+        unsigned numberOfOut = 0;
+
+        PROGLOG("Start loop");
+        for (int i = 0 ; i <= loopCount; i++)
+        {
+            unsigned now = msTick();
+            sem.wait(time);
+            unsigned taken = msTick() - now;
+            sumTaken += taken;
+            maxTaken = (taken > maxTaken ? taken : maxTaken);
+            numberOfOut += (taken > timeLimit ? 1 : 0);
+        }
+
+        VStringBuffer errMsg("values: time: %d, loop: %d, sumTaken: %ld, average taken: %ld, max taken: %d, out of limit: %d times",
+                                time, loopCount, sumTaken, sumTaken/loopCount, maxTaken, numberOfOut );
+        CPPUNIT_ASSERT_MESSAGE(errMsg.str(), 0 == numberOfOut);
+        PROGLOG("%s", errMsg.str());
+    }
+
+    void testSimple()
+    {
+        //Very basic semaphore stress tests.
+        Semaphore sem;
+        sem.signal();
+        sem.wait();
```

Comment:
Right, but probably should have timeout here e.g. if (!sem.wait(1000)) error, better than stalling.
Should never have to wait of course.